### ISSUE TITLE
Custom Embedding layer with Dropout

### ISSUE THREAD

**Issue by riadsouissi** [keras-team/keras#9561]
Hi,
I wanted to implemented a custom dropout in the embedding layer (I am not dropping from the input, instead I am dropping entire words from the embedding dictionary).

I am using call(self, inputs, training) function to override embedding and I need to make sure the dropout is not used during inference.
What happens is that Keras does not pass training=True during the training phase neither passes False during inference (which considering I am using TF backend makes sense since the graph is static).
But I can see that LSTM/GRU layers use the same call() function.
So my question, how do I instruct Keras to pass training mode ? I tried looking at the source code, and honestly, couldn't find a solution...

This is my code:
```
class EmbeddingDropout(Embedding):
    def __init__(self, input_dim, output_dim, word_dropout, **kwargs):
        self.word_dropout = word_dropout #kwargs.get("word_dropout",0)
        super(EmbeddingDropout, self).__init__(input_dim, output_dim, **kwargs)

    def call(self, inputs, training=None):
        print("EmbeddingDropout: training=",training)
        if K.dtype(inputs) != 'int32':
            inputs = K.cast(inputs, 'int32')
            
        if 0 < self.word_dropout < 1:
            _ones = _generate_dropout_ones(self.embeddings, 1) #K.shape(self.embeddings)[-1])
            _dropout_mask = _generate_dropout_mask(_ones,self.word_dropout,training=True)
            _embeddings_with_dropout = self.embeddings * _dropout_mask        
            
        out = K.gather(_embeddings_with_dropout, inputs)
        return out

    def get_config(self):
        config = {'word_dropout': self.word_dropout}
        base_config = super(EmbeddingDropout, self).get_config()
        return dict(list(base_config.items()) + list(config.items()))
```

Thank you!

**Comment by Darthholi** [comment]
I would suggest inspiration from the source -
https://github.com/keras-team/keras/blob/master/keras/layers/core.py
...
where  K.in_train_phase is used :) (for me it works and so I discovered Your issue written here.)

**Comment by riadsouissi** [comment]
Thanks. But it seems K.in_train_phase also expects training parameter from the call() function. My problem is that when call() for my custom embedding is called, it never passes training, so I cannot figure out if I am in inference phase or not.
